web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -0,0 +1,364 @@
+    const agent = agentMeta as DbMeta;
+    integrationService
+      .deployAwsOidcService(agent.integrationName, {
+        deploymentMode: 'database-service',
+        region: agent.selectedAwsRdsDb?.region,
+        subnetIds: agent.selectedAwsRdsDb?.subnets,
+        taskRoleArn,
+        databaseAgentMatcherLabels: labels,
+      })
+      .then(setDeploySvcResp)
+      .catch((err: Error) => {
+        setAttempt({ status: 'failed', statusText: err.message });
+        emitErrorEvent(`deploy request failed: ${err.message}`);
+      });
+  }
+
+  function handleOnProceed() {
+    nextStep(2); // skip the IAM policy view
+    emitEvent(
+      { stepStatus: DiscoverEventStatus.Success }
+      // TODO(lisa) uncomment after backend handles this field
+      // { deployMethod: 'auto' }
+    );
+  }
+
+  function handleDeployFinished(db: Database) {
+    setDeployFinished(true);
+    updateAgentMeta({ ...agentMeta, db, serviceDeployedMethod: 'auto' });
+  }
+
+  function abortDeploying() {
+    if (attempt.status === 'processing') {
+      emitErrorEvent(
+        `aborted in middle of auto deploying (>= 5 minutes of waiting)`
+      );
+    }
+    setDeploySvcResp(null);
+    setAttempt({ status: '' });
+    toggleDeployMethod();
+  }
+
+  const isProcessing = attempt.status === 'processing' && !!deploySvcResp;
+  const isDeploying = isProcessing && !!deploySvcResp;
+  const hasError = attempt.status === 'failed';
+
+  return (
+    <Box>
+      <Validation>
+        {({ validator }) => (
+          <>
+            <Heading
+              toggleDeployMethod={abortDeploying}
+              togglerDisabled={isProcessing}
+            />
+
+            {/* step one */}
+            <CreateAccessRole
+              taskRoleArn={taskRoleArn}
+              setTaskRoleArn={setTaskRoleArn}
+              disabled={isProcessing}
+            />
+
+            {/* step two */}
+            <StyledBox mb={5}>
+              <Text bold>Step 2</Text>
+              <Box mb={4}>
+                <Labels
+                  labels={labels}
+                  setLabels={setLabels}
+                  disableBtns={attempt.status === 'processing'}
+                  showLabelMatchErr={showLabelMatchErr}
+                  dbLabels={dbLabels}
+                  autoFocus={false}
+                />
+              </Box>
+              <ButtonSecondary
+                width="215px"
+                type="submit"
+                onClick={() => handleDeploy(validator)}
+                disabled={attempt.status === 'processing'}
+                mb={2}
+              >
+                Deploy Teleport Service
+              </ButtonSecondary>
+              {hasError && (
+                <Box>
+                  <TextIcon mt={3}>
+                    <Icons.Warning ml={1} color="error.main" />
+                    Encountered Error: {attempt.statusText}
+                  </TextIcon>
+                </Box>
+              )}
+            </StyledBox>
```

Comment:
Could be wrapped into it's own `DefineMatcherLabels` component

Reply:
i could, but didn't think it was so long that it was unreadable so left it as is (there were so many props to pass). if you feel strongly about it i'll pull it out